org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/builder/MavenBuilderImpl.java

```diff
@@ -113,9 +113,8 @@ public Set<IProject> build(MavenSession session, IMavenProjectFacade projectFaca
     Map<Throwable, MojoExecutionKey> buildErrors = new LinkedHashMap<>();
     MavenProjectMutableState snapshot = MavenProjectMutableState.takeSnapshot(mavenProject);
     try {
-      for(Entry<MojoExecutionKey, List<AbstractBuildParticipant>> entry : participants.entrySet()) {
-        MojoExecutionKey mojoExecutionKey = entry.getKey();
-        for(InternalBuildParticipant participant : entry.getValue()) {
+      participants.forEach((mojoExecutionKey, executionParticipants) -> {
```

Comment:
I don't seeing any improvement using foreach with lamda over explicit loop beside its harder to read, harder to debug and produces additional runtime cost.

Reply:
Reverted it here.